RESEARCH/upstream-snapshots/e5b397c9d1883354f03e338dd00f98be3da39f9f/plugins/compound-engineering/skills/ce-compound-refresh/scripts/validate-frontmatter.py

```
@@ -0,0 +1,137 @@
+prevention, not lint.
+
+Pure-stdlib (no PyYAML or other third-party deps). Runs in <50ms typical.
+Designed to produce concrete, actionable error messages so the calling
+agent can fix and retry without ambiguity.
+"""
+import os
+import re
+import sys
+
+
+def usage_fail(msg: str) -> "NoReturn":
```

Ruff (0.15.12)

[error] Undefined name `NoReturn`

(F821)